### PR TITLE
feat: phone auth

### PR DESCRIPTION
**Pull request by FatumaA**
Excuse that i'm still using named routes 🙈 
This is v1 phone auth but Im unable to test with twilio. Please check it out.
Thank you 😊

### CODE REVIEW

**Comment by FatumaA** [comment]
btw, the example app lacks the logic of getting the accessToken from the URL

File: lib/src/screens/supa_phone_auth.dart [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            onPressed: () async {
+              if (!_formKey.currentState!.validate()) {
+                return;
+              }
+
+              try {
+                await supaAuth.signInUserWithPhone(_phone.text, _password.text);
+                successAlert;
+                if (!mounted) return;
+                Navigator.popAndPushNamed(context, widget.redirectUrl, arguments: {"phone": _phone.text});
```
Comment (dshukertjr) [inline lib/src/screens/supa_phone_auth.dart:85]: Should we call the successAlert method here?
```suggestion
                if (!mounted) return;
                await successAlert(context);
                if(mouted) {
                  Navigator.popAndPushNamed(context, widget.redirectUrl, arguments: {"phone": _phone.text});
                }
```

File: lib/src/screens/supa_verify_phone.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            ),
+            onPressed: () async {
+              if (!_formKey.currentState!.validate()) {
+                return;
+              }
+              try {
+                await supaAuth.verifyPhoneUser(data!["phone"], _code.text);
+                successAlert;
+                if (!mounted) return;
+                Navigator.popAndPushNamed(context, widget.redirectUrl ?? '/');
```
Comment (dshukertjr) [inline lib/src/screens/supa_verify_phone.dart:72]: Same here!

File: lib/src/screens/supa_verify_phone.dart [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      autovalidateMode: AutovalidateMode.onUserInteraction,
+      key: _formKey,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          TextFormField(
+            validator: (value) {
+              if (value == null ||
+                  value.isEmpty ) {
+                return 'Please enter a valid email address';
```
Comment (dshukertjr) [inline lib/src/screens/supa_verify_phone.dart:48]:
```suggestion
                return 'Please enter your one time password.';
```

File: lib/src/screens/supa_phone_auth.dart [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    return Form(
+      autovalidateMode: AutovalidateMode.onUserInteraction,
+      key: _formKey,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          TextFormField(
+            validator: (value) {
+              if (value == null || value.isEmpty) {
+                return 'Please enter a valid email address';
```
Comment (dshukertjr) [inline lib/src/screens/supa_phone_auth.dart:45]:
```suggestion
                return 'Please enter a valid phone number';
```

**Review by dshukertjr — COMMENTED** [review]
Thank you so much for the amazing work! I just had few little things, but over all it looks good!

File: lib/src/screens/supa_verify_phone.dart (hunk 3 above)
Comment (FatumaA) [reply lib/src/screens/supa_verify_phone.dart:48]: I think its clearer to say enter the code sent than a password, just to minimise possible confusion for users.

File: lib/src/screens/supa_phone_auth.dart (hunk 1 above)
Comment (FatumaA) [reply lib/src/screens/supa_phone_auth.dart:85]: I was thinking tell them it was successful before redirecting them to where they enter the code sent to the number. What were your thoughts?
Also, do we need to explicitly check for mounted?

Comment (dshukertjr) [reply lib/src/screens/supa_phone_auth.dart:85]: I agree with you! 

This doesn't show the dialog though, right?
```dart
successAlert;
```

>Also, do we need to explicitly check for mounted?
According to the new lint rule we do, but we can ignore them if you want to.

File: lib/src/screens/supa_verify_phone.dart (hunk 3 above)
Comment (dshukertjr) [reply lib/src/screens/supa_verify_phone.dart]: Sure, I am not attached to this particular line at all, so please feel free to change it to whatever you think is best!

**Comment by FatumaA** [comment]
@dshukertjr Hi, I made the suggested changes

File: lib/src/screens/supa_phone_auth.dart [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            ),
+            onPressed: () async {
+              if (!_formKey.currentState!.validate()) {
+                return;
+              }
+
+              try {
+                await supaAuth.signInUserWithPhone(_phone.text, _password.text);
+                if (!mounted) return;
+                successAlert;
```
Comment (dshukertjr) [inline lib/src/screens/supa_phone_auth.dart:84]: Calling the method here!
```suggestion
                await successAlert(context);
```

File: lib/src/screens/supa_verify_phone.dart [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+              style: TextStyle(fontWeight: FontWeight.bold),
+            ),
+            onPressed: () async {
+              if (!_formKey.currentState!.validate()) {
+                return;
+              }
+              try {
+                await supaAuth.verifyPhoneUser(data!["phone"], _code.text);
+                if (!mounted) return;
+                successAlert;
```
Comment (dshukertjr) [inline lib/src/screens/supa_verify_phone.dart:69]: And here!
```suggestion
                await successAlert(context);
```

File: lib/src/screens/supa_phone_auth.dart [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+              prefixIcon: Icon(Icons.phone),
+              hintText: 'Enter your phone number',
+            ),
+            controller: _phone,
+          ),
+          spacer(16),
+          TextFormField(
+            validator: (value) {
+              if (value == null || value.isEmpty || value.length < 6) {
+                return 'Please enter a password that is atleast 6 characters long';
```
Comment (dshukertjr) [inline lib/src/screens/supa_phone_auth.dart:59]:
```suggestion
                return 'Please enter a password that is at least 6 characters long';
```

**Review by dshukertjr — APPROVED** [review]
LGTM! Thanks for the amazing work!

**Pull request merged**